src/vga/vga_buffer.rs

```diff
@@ -63,10 +71,13 @@ impl PlainDrawable for VGADevice<'_> {
             ),
         }
     }
+    fn draw_point_p(&mut self, p: Point2D<u16>, color: VGAColor<u8>) {
+        self.draw_point(p.x, p.y, color);
+    }
 
-    fn draw_line(&mut self, x1: usize, y1: usize, x2: usize, y2: usize, color: &VGAColor) {
-        let ix2: isize = x2.try_into().unwrap();
-        let iy2: isize = y2.try_into().unwrap();
+    fn draw_line(&mut self, x1: u16, y1: u16, x2: u16, y2: u16, color: VGAColor<u8>) {
+        let ix2: isize = x2 as isize;
```

Comment:
whats the reason for using `isize` over `usize`

Reply:
dx and dy are signed, so x and y have to be signed, so I have converted those two to signed too, so I don't need to do this for each comparison. The fact I can reassign variables means I can make this leaner

Reply:
ah, fair enough